packages/eslint-plugin/src/util/isNodeEqual.ts

```diff
@@ -0,0 +1,77 @@
+ *   AssignmentPattern AwaitExpression BinaryExpression BlockStatement
+ *   BreakStatement CallExpression CatchClause ChainExpression ClassBody
+ *   ClassDeclaration ClassExpression ConditionalExpression ContinueStatement
+ *   DebuggerStatement Decorator DoWhileStatement EmptyStatement
+ *   ExportAllDeclaration ExportDefaultDeclaration ExportNamedDeclaration
+ *   ExportSpecifier ExpressionStatement ForInStatement ForOfStatement
+ *   ForStatement FunctionDeclaration FunctionExpression IfStatement
+ *   ImportAttribute ImportDeclaration ImportDefaultSpecifier ImportExpression
+ *   ImportNamespaceSpecifier ImportSpecifier JSXAttribute JSXClosingElement
+ *   JSXClosingFragment JSXElement JSXEmptyExpression JSXExpressionContainer
+ *   JSXFragment JSXIdentifier JSXMemberExpression JSXNamespacedName
+ *   JSXOpeningElement JSXOpeningFragment JSXSpreadAttribute JSXSpreadChild
+ *   JSXText LabeledStatement LogicalExpression MetaProperty MethodDefinition
+ *   NewExpression ObjectExpression ObjectPattern PrivateIdentifier Program
+ *   Property PropertyDefinition RestElement ReturnStatement SequenceExpression
+ *   SpreadElement StaticBlock Super SwitchCase SwitchStatement
+ *   TaggedTemplateExpression TemplateElement TemplateLiteral ThrowStatement
+ *   TryStatement TSAbstractKeyword TSAbstractMethodDefinition
+ *   TSAbstractPropertyDefinition TSAnyKeyword TSArrayType TSAsExpression
+ *   TSAsyncKeyword TSBigIntKeyword TSBooleanKeyword TSCallSignatureDeclaration
+ *   TSClassImplements TSConditionalType TSConstructorType
+ *   TSConstructSignatureDeclaration TSDeclareFunction TSDeclareKeyword
+ *   TSEmptyBodyFunctionExpression TSEnumDeclaration TSEnumMember
+ *   TSExportAssignment TSExportKeyword TSExternalModuleReference
+ *   TSFunctionType TSImportEqualsDeclaration TSImportType TSIndexedAccessType
+ *   TSIndexSignature TSInferType TSInterfaceBody TSInterfaceDeclaration
+ *   TSInterfaceHeritage TSIntersectionType TSIntrinsicKeyword TSLiteralType
+ *   TSMappedType TSMethodSignature TSModuleBlock TSModuleDeclaration
+ *   TSNamedTupleMember TSNamespaceExportDeclaration TSNeverKeyword
+ *   TSNonNullExpression TSNullKeyword TSNumberKeyword TSObjectKeyword
+ *   TSOptionalType TSParameterProperty TSPrivateKeyword TSPropertySignature
+ *   TSProtectedKeyword TSPublicKeyword TSQualifiedName TSReadonlyKeyword
+ *   TSRestType TSStaticKeyword TSStringKeyword TSSymbolKeyword
+ *   TSTemplateLiteralType TSThisType TSTupleType TSTypeAliasDeclaration
+ *   TSTypeAnnotation TSTypeAssertion TSTypeLiteral TSTypeOperator
+ *   TSTypeParameter TSTypeParameterDeclaration TSTypeParameterInstantiation
+ *   TSTypePredicate TSTypeQuery TSTypeReference TSUndefinedKeyword TSUnionType
+ *   TSUnknownKeyword TSVoidKeyword UnaryExpression UpdateExpression
+ *   VariableDeclaration VariableDeclarator WhileStatement WithStatement
+ *   YieldExpression
+ */
+
+export function isNodeEqual(a: TSESTree.Node, b: TSESTree.Node): boolean {
```

Comment:
I will think about this one.

Reply:
I have thought about this, I don't like `doNodesContainSameValue` since we don't just compare the contents or value.

I however also have no idea for a better name 🤷‍♂️